plugins/modules/rds_instance_info.py

```diff
@@ -377,11 +377,24 @@ def _describe_db_instances(conn, **params):
     return results
 
 
-def instance_info(module, conn):
-    instance_name = module.params.get('db_instance_identifier')
-    filters = module.params.get('filters')
+class RdsInstanceInfoFailure(Exception):
```

Comment:
I do not really understand why we should use this custom exceptions class.

Reply:
When it comes to unit testing, it's much easier to test a function which simply raises an error than mocking up a "module" and handling the fail_json / fail_json_aws calls

Reply:
Ok, so we're going to uniform that for the other modules as well (I mean those we're willing to cover by unit).

Reply:
Compare the complexity of 
https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/plugins/modules/ec2_instance/test_determine_iam_role.py#L50

```
pytest.raises(SomeException):
   do_something_which_fails()
```

The down side is that you need to think a little about how to add any extra state to the exception so that it can be made available to the eventual
```
except SomeException:
  module.fail_json_aws(....)
```